PathOfBuildingCommunity/PathOfBuilding
# Add support for inc brand aoe after 50% attached duration mastery

### Description of the problem being solved:
Adds support for the "Brands have 30% increased Area of Effect if 50% of the Attached Duration expired" brand mastery.

Adds a new config "Last 50% of attached duration?", similar to the current 25% from from the heiro ascendancy.

AoE only applies to brands -> had to introduce small changes to the AoE calc to support brand specific AoE modifiers. The inc. AoE mod is still shown in the breakdown for non-brands but it is not applied. If s1 has a clue on how to have a skill type condition in the CalcSections.lua feel free to comment.
E.g. Toxic Rain:
![image](https://github.com/user-attachments/assets/1bf6d765-acf4-430b-8c06-16c8612e1dd7)

### Steps taken to verify a working solution:
- Select a brand skill with a radius (e.g. arma brand)
- Spec the 30% aoe if 50% expired brand mastery
- Enable the "Last 50% of attached duration?" config
- AoE calcs should have 30% inc aoe for all radii (primary, secondary, tertiary).
- If a normal skill is selected then the AoE will not apply (e.g. toxic rain)

### Link to a build that showcases this PR:
https://pobb.in/UTuU9CKlmVuY

### Before screenshot:
![image](https://github.com/user-attachments/assets/021a01d6-f8e7-4d45-bce5-96498da3e801)

### After screenshot:
![image](https://github.com/user-attachments/assets/97a2ee17-da83-49e1-be2c-ec380aefbd49)
![image](https://github.com/user-attachments/assets/8dcf3ace-696b-4144-a751-6c66fcc3b44d)
